### PR TITLE
CIL-414: external variable openshift_persist_registry

### PR DESCRIPTION
**Pull request by mdanter**
this controls that:
1. ec2 instance userdata for infra nodes optionally creates a directory with the correct permissions via oadm
2. the master set up the registry in scc and  creates/modifies the docker registry pod

This is tested with and without specifying openshift_hosted_manage_registry=false, and works.

### CODE REVIEW

File: playbooks/aws/openshift-cluster/terminate.yml [hunk 1]
```diff
@@ -40,8 +40,8 @@
         tags:
           environment:   "{{ hostvars[item]['ec2_tag_environment'] }}"
           clusterid:     "{{ hostvars[item]['ec2_tag_clusterid'] }}"
-          host-type:     "{{ hostvars[item]['ec2_tag_host-type'] }}"
```
Comment (oybed) [inline playbooks/aws/openshift-cluster/terminate.yml:43]: Why these changes? Are these bugs?

File: playbooks/aws/openshift-cluster/vars.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ deployment_rhel7_ent_base:
   # rhel-7.1, requires cloud access subscription
   image: "{{ lookup('oo_option', 'ec2_image') | default('ami-10251c7a', True) }}"
   image_name: "{{ lookup('oo_option', 'ec2_image_name') | default(None, True) }}"
-  region: "{{ lookup('oo_option', 'AWS_REGION') | default('us-east-1', True) }}"
```
Comment (oybed) [inline playbooks/aws/openshift-cluster/vars.yml:8]: Did you mean to revert these, is this a merge issue with the previous commits?

File: playbooks/aws/openshift-cluster/terminate.yml (hunk 1 above)
Comment (mdanter) [inline playbooks/aws/openshift-cluster/terminate.yml:43]: yes, they were bugs that I caught when testing this with tower.

File: playbooks/aws/openshift-cluster/vars.yml (hunk 2 above)
Comment (mdanter) [inline playbooks/aws/openshift-cluster/vars.yml:8]: Yes, I did mean to revert this. When testing this with ansible tower, it became apparent why they were using ec2-region.

**Comment by oybed** [comment]
@mdanter Looks good - merging
